inc/class-statify-frontend.php

```diff
@@ -91,15 +91,24 @@ public static function track_visit( $is_snippet = false ) {
 			$data['referrer'] = esc_url_raw( $referrer, array( 'http', 'https' ) );
 		}
 
-		/* Relative target url */
+		// Relative target URL.
 		$data['target'] = user_trailingslashit( str_replace( home_url( '/', 'relative' ), '/', $target ) );
 
-		// Trim target url.
+		// Trim target URL, i.e. remove query parameters.
 		if ( $wp_rewrite->permalink_structure ) {
-			$data['target'] = wp_parse_url( $data['target'], PHP_URL_PATH );
+			$parsed_target  = wp_parse_url( $data['target'] );
+			$data['target'] = $parsed_target['path'];
+
+			/*
+			 * Re-add AMP parameter to keep that information (only applicable for JS tracking).
+			 * We can assume "amp" to be the only parameter, as ${canonicalPath} already eliminates other queries.
+			 */
+			if ( $is_snippet && isset( $parsed_target['query'] ) && 'amp/' === $parsed_target['query'] ) {
+				$data['target'] .= '?amp';
```

Comment:
What about using `add_query_arg` even if the query string should be empty?

Reply:
Using `add_query_arg( 'amp', '', $data['target'] )` would not hurt at this point.

As far as I remember query arguments should never be part of canonical paths, but we never know.